app/src/androidTest/java/ch/epfl/polybazaar/FillListingActivityTest.java

```diff
@@ -240,6 +248,63 @@ public void cancelTakingPicture() throws Throwable {
     }
 
 
+    @Test
+    public void DialogAppearsWhenNoConnection() throws Throwable {
+        Intents.init();
+        useMockNetworkState(false);
+        useMockDataStore();
+        fillListing();
+        runOnUiThread(() -> fillSaleActivityTestRule.getActivity().findViewById(R.id.submitListing).performClick());
+        Thread.sleep(5000);
+        assert(fillSaleActivityTestRule.getActivity().getSupportFragmentManager().findFragmentByTag("noConnectionDialog").isVisible());
+        //onView(withText("No Internet connection found")).check(matches(isDisplayed()));
+        Intents.release();
+        useRealNetwork();
+    }
+
+    @Test
+    public void DialogPositiveClickGoesToSalesOverview() throws Throwable {
+        Intents.init();
+        useMockNetworkState(false);
+        useMockDataStore();
+        fillListing();
+        runOnUiThread(() -> fillSaleActivityTestRule.getActivity().findViewById(R.id.submitListing).performClick());
+        Thread.sleep(5000);
+        runOnUiThread(() -> {
+            DialogFragment dialogFragment = (DialogFragment)fillSaleActivityTestRule.getActivity().getSupportFragmentManager().findFragmentByTag("noConnectionDialog");
+            AlertDialog dialog = (AlertDialog) dialogFragment.getDialog();
+            Button posButton = dialog.getButton(DialogInterface.BUTTON_POSITIVE);
+            posButton.performClick();
+        });
+        Thread.sleep(5000);
+        intended(hasComponent(SalesOverview.class.getName()));
+        Intents.release();
+        useRealNetwork();
+    }
+
+    @Test
+    public void DialogNegativeClickGoesToFillListing() throws Throwable {
+        Intents.init();
+        useMockNetworkState(false);
+        useMockDataStore();
+        fillListing();
+        runOnUiThread(() -> fillSaleActivityTestRule.getActivity().findViewById(R.id.submitListing).performClick());
+        Thread.sleep(5000);
+        runOnUiThread(() -> {
+
+            DialogFragment dialogFragment = (DialogFragment)fillSaleActivityTestRule.getActivity().getSupportFragmentManager().findFragmentByTag("noConnectionDialog");
+            AlertDialog dialog = (AlertDialog) dialogFragment.getDialog();
+            Button negButton = dialog.getButton(DialogInterface.BUTTON_NEGATIVE);
+            negButton.performClick();
+        });
+        //onView(withText("Cancel")).perform(click());
+        Thread.sleep(5000);
+        hasComponent(FillListingActivity.class.getName());
+        Intents.release();
+        useRealNetwork();
```

Comment:
Do you really have to call useRealNetwork at the end of a test ?

In general, all the calls to useMockDataStore and similar methods could be done in a @BeforeClass and @AfterClass